### PR TITLE
Replace hardcoded ids with hook

### PR DESCRIPTION
**Pull request by rafawendel**
## What type of PR is this? (check all applicable)

- [X] Refactor

## Description
This replaces all hard coded `id` attributes with an unique ID generated from `useUniqueId`.
The code base was searched for all occurrences of the previous IDs to avoid unwanted lost references.

### CODE REVIEW

File: client/app/components/CreateSourceDialog.jsx [hunk 1]
```diff
@@ -45,6 +46,8 @@ class CreateSourceDialog extends React.Component {
     currentStep: StepEnum.SELECT_TYPE,
   };
 
+  formId = useUniqueId("sourceForm");
```
Comment (kravets-levko) [inline client/app/components/CreateSourceDialog.jsx:49]: Just wondering: how does it work with a class component? Hooks shouldn't work with class components, only with functional ones

Comment (rafawendel) [reply client/app/components/CreateSourceDialog.jsx:49]: I didn't even think about it Lol. Will workaround it soon.

**Comment by thielium** [comment]
Will check it out after the class components issue is resolved, but so far LGTM

**Comment by rafawendel** [comment]
UPDATE: the problem sorted itself. Blocked by #5474 -- will be rebased.

Ignore import order, will add `eslint-plugin-imports` soon.

File: client/app/pages/dashboards/DashboardPage.jsx [hunk 2]
```diff
@@ -144,7 +147,7 @@ function DashboardComponent(props) {
         </div>
       )}
       {editingLayout && <DashboardSettings dashboardConfiguration={dashboardConfiguration} />}
-      <div id="dashboard-container">
+      <div id={dashboardContainerId}>
```
Comment (kravets-levko) [inline client/app/pages/dashboards/DashboardPage.jsx:150]: This one is different - it's used for styling:
https://github.com/getredash/redash/blob/d8d7c78992e44a4b6d7fdd4c39ccc1c35cd8c7a9/client/app/pages/dashboards/DashboardPage.less#L16-L19

File: client/app/pages/dashboards/PublicDashboardPage.jsx [hunk 3]
```diff
@@ -37,7 +40,7 @@ function PublicDashboard({ dashboard }) {
           <Filters filters={filters} onChange={setFilters} />
         </div>
       )}
-      <div id="dashboard-container">
+      <div id={dashboardContainerId}>
```
Comment (kravets-levko) [inline client/app/pages/dashboards/PublicDashboardPage.jsx:43]: Same as above: https://github.com/getredash/redash/blob/d8d7c78992e44a4b6d7fdd4c39ccc1c35cd8c7a9/client/app/pages/dashboards/DashboardPage.less#L16-L19

File: package.json [hunk 4]
```diff
@@ -73,6 +73,7 @@
     "react-grid-layout": "^0.18.2",
     "react-resizable": "^1.10.1",
     "react-virtualized": "^9.21.2",
+    "short-uuid": "^4.1.0",
```
Comment (kravets-levko) [inline package.json:76]: 👍

**Review by thielium — COMMENTED** [review]
General question / comment: this approach is going to create a lot of thrashing of ids / updating the DOM, right? Every single render, for some of these usages, we get a new id.

What's the motivation, here? Are we worried about collisions? What was wrong with the `useUniqueId` approach? I've seen that in other codebases, and it works well. At the minimum, we should figure out a way to make the uuids stable.

**Comment by kravets-levko** [comment]
> General question / comment: this approach is going to create a lot of thrashing of ids / updating the DOM, right? Every single render, for some of these usages, we get a new id.

> What's the motivation, here? Are we worried about collisions? What was wrong with the useUniqueId approach? I've seen that in other codebases, and it works well. At the minimum, we should figure out a way to make the uuids stable.

The motivation is to keep things simple. We can always wrap `getUniqueId()` calls with `useMemo` if it makes any problems.

Re. collisions: `short-uuid` generates a regular 16-bytes UUIDs, but uses base64-like representation instead of hexadecimal. There's practically no chance to get a collision.

**Review by kravets-levko — APPROVED** [review]
Please merge it before you decide to change it back :smile:

**Comment by thielium** [comment]
>The motivation is to keep things simple.

I think we both want this!

Might be worth an offline discussion.

**Pull request merged**